src/key-generator.js

```diff
@@ -104,6 +104,10 @@ KeyGenerator.prototype._generateNewKeys = function _generateNewKeys() {
 					this._keyGenerationTask = undefined;
 					return this._currentPrivateKey;
 				});
+		})
+		.catch(() => {
+			return new Promise(resolve => setTimeout(resolve, 100).unref())
```

Comment:
This fix would help! To get to the next level though, we could leverage something like https://www.npmjs.com/package/promise-retry? (disclaimer: just a search result, haven't used it).